### PR TITLE
Fix allow list with multiple license

### PR DESCRIPTION
**Pull request by kachick**
I have faced an issue to integrate [stylelint](https://github.com/stylelint/stylelint). [It depends meow](https://github.com/stylelint/stylelint/blob/f48c1d1518d6429c7516ae73f3d81e452f57287a/package.json#L132), [meow depends type-fest](https://github.com/sindresorhus/meow/blob/85dc1e9c35f7844941690740c68b839691bd515f/package.json#L55). Then [type-fest has dual license](https://github.com/sindresorhus/type-fest/blob/2f418dbbb6182c53cdac26dc3421f7e8806c789b/package.json#L5).

Their license is written in SPDX 2.0, the syntax is [recommended by npm](https://github.com/npm/cli/blob/facba42bcd2a10722788f44de13ea46cde6c8e71/docs/content/configuring-npm/package-json.md?plain=1#L125-L127).

GitHub REST API returns as below.

```
  {
    "change_type": "added",
    "manifest": "package-lock.json",
    "ecosystem": "npm",
    "name": "type-fest",
    "version": "0.18.1",
    "package_url": "pkg:npm/type-fest@0.18.1",
    "license": "CC0-1.0 OR MIT OR (CC0-1.0 AND MIT)",
    "source_repository_url": "https://github.com/sindresorhus/type-fest",
    "vulnerabilities": [

    ]
  },
  {
    "change_type": "added",
    "manifest": "package-lock.json",
    "ecosystem": "npm",
    "name": "type-fest",
    "version": "0.8.1",
    "package_url": "pkg:npm/type-fest@0.8.1",
    "license": "MIT OR (CC0-1.0 AND MIT)",
    "source_repository_url": "https://github.com/sindresorhus/type-fest",
    "vulnerabilities": [

    ]
  },
  {
    "change_type": "added",
    "manifest": "package-lock.json",
    "ecosystem": "npm",
    "name": "type-fest",
    "version": "0.6.0",
    "package_url": "pkg:npm/type-fest@0.6.0",
    "license": "MIT OR (CC0-1.0 AND MIT)",
    "source_repository_url": "https://github.com/sindresorhus/type-fest",
    "vulnerabilities": [

    ]
  },
```

That is not simple to parse with current parsing logics.
So I have added the library of https://github.com/jslicense/spdx-satisfies.js.

Although, `deny list` looks hard to integrate to current spec. (And guessing hard to implement by me 🙇‍♂️ )
So I have added skipping test only for deny list. If it is not good, please remove.

### CODE REVIEW

**Comment by kachick** [comment]
Run eslint with existing config makes many non related alerts. I have updated them. However when the diff is needless, I'll revert them. (Looks CI is not checking them) 🙏

**Comment by brphelps** [comment]
@kachick -- I'm not super familiar with the licensing changes, but it looks like you've grouped together some unrelated changes like adding linting + linter integration. It seems like those changes warrant a PR split!

File: __tests__/config.test.ts [hunk 1]
```diff
@@ -26,11 +26,11 @@ test('it defaults to low severity', async () => {
 
 test('it reads custom configs', async () => {
   setInput('fail-on-severity', 'critical')
-  setInput('allow-licenses', ' BSD, GPL 2')
+  setInput('allow-licenses', ' BSD-2-Clause, GPL-2.0')
```
[the remark this answers is not in the document]
Comment (kachick) [reply __tests__/config.test.ts:29]: Added kind error into early stage https://github.com/actions/dependency-review-action/pull/131/commits/6738e7d378137fb002c35eda956e37ea67afc9e6

**Comment by kachick** [comment]
@brphelps Sure! I have updated this PR for focusing multiple license issue.

And extracted CI and lint in following PR https://github.com/kachick/dependency-review-action/pull/1
I can send it after this PR closed.

**Comment by febuiles** [comment]
@kachick Thanks a ton for highlighting this issue! I need to find out how the licenses are being handled in the backend, and see if it's expected that they will include SPDX expressions going forward. I'll post an update when I have more info!

**Comment by Baro1905** [comment]
- name: Dependency Review
  uses: actions/dependency-review-action@v2.0.2

**Comment by Baro1905** [comment]
kachick:multiple-licenses

**Comment by febuiles** [comment]
@kachick Going forward the API will be using SPDX licenses. I have some ideas about augmenting our current license dataset, but for the time being I think using `spdx-satisfies` is the proper way to move forward. What conflicts did you encounter when dealing with the denylist?

**Comment by kachick** [comment]
I couldn’t imagine reasonable specs for a denying condition and a given combining condition. 🤔 

For example, below patterns.

Allow License | Deny License | Given License | They should be…?
-- | -- | -- | --
MIT | GPL-1.0-or-later | GPL-1.0-or-later AND MIT | False
MIT | GPL-1.0-or-later | GPL-1.0-or-later OR MIT | Validation error
(Empty list is given) | GPL-1.0-or-later | GPL-1.0-or-later OR MIT | **True?**


Personally I hope false-positive is the preferable way for this action. Or raising errors for some patterns might be robust even if making many alerts.

**Comment by febuiles** [comment]
Thanks for bringing up a good point! I will remove the example with an allow list since the action doesn't support both allow/deny: https://github.com/actions/dependency-review-action/#licenses. Using an example from an actual library:

Deny License | Given License | Result
-- | -- | --
MIT/X11 OR Apache-2.0 | MIT/X11 | Invalid license
MIT/X11 | MIT/X11 OR Apache-2.0 | Valid license

I've thought about this for a while, and the SPDX expression is valid, so we should not be biasing what happens here. 

I was initially a bit concerned about SDPX expressions, but I've noticed that they are used more to express "This project uses license X AND Y" than "This project has uses X AND has used Y in the past". 

I fetched a full list of unique licenses for all repos: https://gist.github.com/febuiles/985fb5215e72fd196cd9a7ea7fca1a2f. You can see there's a lot of funky stuff in there, but the actual number of repos with those long licenses is not significant.

@kachick what do you make out of all of this? Does it make sense to use `spdx-satisfies` to cover both allow/deny lists under this lens?

**Pull request closed without merging**